MudSelect/MudDatePicker: Fix accidental MudDatePicker break while maintaining desired MudSelect behavior

Anu6is opened the pull request:
## Description
#10215 set the min-width of the popover to match the parent container as long as `mud-popover-relative-width` class was not present. This has unintended side effects for other components (MudDatePicker) that didn't utilize the `relative` class but also didn't need to match the parent container. 

This change introduces an explicit class `mud-popover-adaptive-width` that should be used if popover should, at a minimum, match the parent class but isn't restricted by (can exceed) the width of the parent class. 

**RelativeWidth** has been made nullable
- null: do nothing (min/max width are not defined)
- true: restricts the max-width to the parent container width
- false: restricts the min-width to the parent container width  

Resolves secondary issue noticed in #10235 - does not solve the actual reported issue

## How Has This Been Tested?
- Visually 
- Existing unit tests

## Type of Changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (fix or improvement to the website or code docs)

## Checklist
- [x] The PR is submitted to the correct branch (`dev`).
- [x] My code follows the code style of this project.
- [x] I've added relevant tests.

Anu6is committed:
MudSelect - add explict class for adaptive width

Anu6is committed:
Update MudSelect.razor.cs

Anu6is committed:
Merge branch 'MudBlazor:dev' into DatePicker-Break

danielchalmers reviewed (changes requested):
I would move the classname logic out of MudSelect and into MudPopover so it can be reused for other controls

Aka

1. Delete/revert `PopoverClassname`
2. Add a new property `MudPopover.AdaptiveWidth`
3. Add `.AddClass($"mud-popover-adaptive-width", AdaptiveWidth)` to the popover

Anu6is commented:
> I would move the classname logic out of MudSelect and into MudPopover so it can be reused for other controls
> 
> Aka
> 
> 1. Delete/revert `PopoverClassname`
> 2. Add a new property `MudPopover.AdaptiveWidth`
> 3. Add `.AddClass($"mud-popover-adaptive-width", AdaptiveWidth)` to the popover

That was my initial thought, but I didn't really like mapping `FullWidth` from the Select to two separate parameters in the Popover. It not becomes 
```html
<MudPopover RelativeWidth="FullWidth" AdaptiveWidth="!FullWidth">  
```

And now we have two properties that are currently mutually exclusive that could end being set at the same time (although I guess it doesn't have to be mutually exclusive). If you're cool with both properties mapping to `FullWidth` thought, I'll go ahead and make that change.

danielchalmers commented:
> > I would move the classname logic out of MudSelect and into MudPopover so it can be reused for other controls
> > Aka
> > 
> > 1. Delete/revert `PopoverClassname`
> > 2. Add a new property `MudPopover.AdaptiveWidth`
> > 3. Add `.AddClass($"mud-popover-adaptive-width", AdaptiveWidth)` to the popover
> 
> That was my initial thought, but I didn't really like mapping `FullWidth` from the Select to two separate parameters in the Popover. It now becomes
> 
> ```
> <MudPopover RelativeWidth="FullWidth" AdaptiveWidth="!FullWidth">  
> ```
> 
> And now we have two properties that are currently mutually exclusive that could end being set at the same time (although I guess it doesn't have to be mutually exclusive). If you're cool with both properties mapping to `FullWidth` thought, I'll go ahead and make that change.

Are there any cases in our components where we would want to enable both relative width and adaptive width, or disable them both? Sorry haven't been able to keep up 100% with the changes

Anu6is commented:
Binding to one is what is does now ... `RelativeWidth` becomes either true or false based on FullWidth. But given the regression it caused, MudPicker doesn't want either min or max to be set (both should be disabled). Can't say if any other component is like that.

danielchalmers commented:
I'm okay with whatever you think makes sense, I just believe the logic should be done in the popover's code instead of the consumer

Anu6is committed:
make popover RelativeWidth nullable.
- when null - do nothing
- when true - set max width
- when false - set min width

Anu6is committed:
Merge branch 'dev' into DatePicker-Break

danielchalmers reviewed (approved):
Smart

ScarletKuro commented:
Should this be added to breaking change? @henon since `RelativeWidth` is now nullable?

ScarletKuro commented:
Added to migration guide https://github.com/MudBlazor/MudBlazor/issues/9953 just in case.

Ty.
